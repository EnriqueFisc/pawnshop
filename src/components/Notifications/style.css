.notification__container {
    box-sizing: border-box;
    top: 10px;
    right: 10px;
    position: absolute;
    width: 350px;
    height: 80px;
    
    color: white;
    padding: 10px;
}
.error {
    background-color: #F55D56;
    border: 1px solid #F51C44;
}
.success {
    background-color: #61F591;
    border: 1px solid #05F54B;
}