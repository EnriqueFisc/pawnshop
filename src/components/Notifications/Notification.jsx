import React, { useEffect } from "react";
import "./style.css";

export const Notification = ({ msg, ok, setNotification }) => {
  useEffect(() => {
    const interval = setInterval(() => {
      setNotification({
        load: false,
        msg: "",
      });
    }, 5000);

    return () => {
      clearInterval(interval);
    };
  }, []);

  return (
    <div className={`notification__container ${ok ? "success" : "error"}`}>
      <p>Error:</p>
      <span>{msg}</span>
    </div>
  );
};
