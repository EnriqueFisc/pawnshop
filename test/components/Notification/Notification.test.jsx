import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { Notification } from "../../../src/components/Notifications/Notification";

describe("Test in Notification Component", () => {
  test("should match with snapshot", () => {
    const component = render(
      <Notification msg={"Testing"} ok={false} setNotification={() => {}} />
    );
    expect(component.container).toMatchSnapshot();
  });
});
