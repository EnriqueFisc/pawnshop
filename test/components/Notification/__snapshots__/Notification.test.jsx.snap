// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test in Notification Component > should match with snapshot 1`] = `
<div>
  <div
    class="notification__container error"
  >
    <p>
      Error:
    </p>
    <span>
      Testing
    </span>
  </div>
</div>
`;
